runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -653,6 +659,9 @@ private static void setUpWorkLoggingContext(String workId, String computationId)
   }
 
   private int chooseMaximumNumberOfThreads() {
+    if (maxThreadCountOverride.get() != 0) {
```

Comment:
since this is possibly modified, it's safer to save the value you observe instead of observing it again

probably wouldn't ever actually change between calls but might as well be safe.